pkg/cnb/remote_stack_reader.go

```diff
@@ -25,13 +25,17 @@ type RemoteStackReader struct {
 	Keychain       authn.Keychain
 }
 
-func (r *RemoteStackReader) Read(clusterStackSpec v1alpha1.ClusterStackSpec) (v1alpha1.ResolvedClusterStack, error) {
-	buildImage, buildIdentifier, err := r.RegistryClient.Fetch(r.Keychain, clusterStackSpec.BuildImage.Image)
+func (r *RemoteStackReader) Read(keychain *authn.Keychain, clusterStackSpec v1alpha1.ClusterStackSpec) (v1alpha1.ResolvedClusterStack, error) {
+	if keychain == nil {
+		keychain = &r.Keychain
```

Comment:
Now that we always generate the keychain can we drop the keychain injected in here?

Reply:
I think originally we injected the keychain as a bit of a shortcut that now is unnecessary. The keychain factory accepts an empty `SecretRef` which we use to [create the injected keychain](https://github.com/pivotal/kpack/blob/3a6c993d4fa82cd310f2ba2d84677ecfb08921fd/cmd/controller/main.go#L144). To make this flow follow the same pattern as the builder creation we can always create a keychain in the store/stack reconciler and provide it as an argument to the StackReader. If no ServiceAccountRef is specified on the ClusterStack or ClusterStore the keychain factory can be invoked with an empty `SecretRef`.

Reply:
>...To make this flow follow the same pattern as the builder creation we can always create a keychain in the store/stack reconciler and provide it as an argument to the StackReader. If no ServiceAccountRef is specified on the ClusterStack or ClusterStore the keychain factory can be invoked with an empty `SecretRef`.

You are right, I think that makes the code easier to understand. Could you have a look if I understood you correct?